tests/graph.rs

```diff
@@ -342,6 +343,62 @@ fn dijk() {
     assert_eq!(scores[&c], 9);
 }
 
+#[test]
+fn test_astar_null_heuristic() {
+    let mut g = Graph::new();
+    let a = g.add_node("A");
+    let b = g.add_node("B");
+    let c = g.add_node("C");
+    let d = g.add_node("D");
+    let e = g.add_node("E");
+    let f = g.add_node("F");
+    g.add_edge(a, b, 7);
+    g.add_edge(c, a, 9);
+    g.add_edge(a, d, 14);
+    g.add_edge(b, c, 10);
+    g.add_edge(d, c, 2);
+    g.add_edge(d, e, 9);
+    g.add_edge(b, f, 15);
+    g.add_edge(c, f, 11);
+    g.add_edge(e, f, 6);
+
+    let path = astar(&g, a, |finish| finish == e, |e| *e.weight(), |_| 0);
+    assert_eq!(path, Some((23, vec![a, d, e])));
+
+    let path = astar(&g, e, |finish| finish == b, |e| *e.weight(), |_| 0);
+    assert_eq!(path, None);
+}
```

Comment:
Looks like a good enough test. The exhaustive approach is to use dijkstra and check that every path and path cost for *a* to *any node* agrees in both algorithms with the trivial heuristic.

We've had success with the exhaustive approach, checking algorithms against each other in petgraph before. It's a good way to gain confidence in the implementation. (See bugs in this crate's history)

Reply:
That kind of setup https://github.com/bluss/petgraph/pull/61